pkg/controller/core/workload_controller.go

```diff
@@ -472,6 +494,38 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{}, nil
 }
 
+func (r *WorkloadReconciler) finalize(ctx context.Context, wl *kueue.Workload, log logr.Logger) error {
+	log.V(2).Info("Finalizing workload.")
+	defer r.notifyWatchers(wl, nil)
+
+	if workload.HasQuotaReservation(wl) {
+		var err error = nil
```

Comment:
You don't need `= nil` since `error` is zero-initialized to nil:
```suggestion
		var err error
```